### PR TITLE
feat(wallet): Wired Up SelectedCoin in UI

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Wires up `getSelectedCoin` and `setSelectedCoin` in the Wallet UI
This also allowed to fix a bug where the `SelectedAccount` and `SelectedNetwork` were not updating between the `Panel` and the `Page.

Resolves <https://github.com/brave/brave-browser/issues/22820>
Resolves <https://github.com/brave/brave-browser/issues/22564>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

https://user-images.githubusercontent.com/40611140/168657359-2373ecb0-b19d-4bb6-85a1-4067d59b0253.mov

### CODE REVIEW

File: components/brave_wallet_ui/common/async/handlers.ts [hunk 1]
```diff
@@ -106,6 +110,31 @@ async function updateAccountInfo (store: Store) {
   }
 }
 
+async function updateCoinAccountNetworkInfo (store: Store, coin: BraveWallet.CoinType) {
+  const state = getWalletState(store)
+  const { accounts, networkList } = state
```
Comment (josheleonard) [inline components/brave_wallet_ui/common/async/handlers.ts:115]: can be combined with line above

File: components/brave_wallet_ui/common/async/handlers.ts [hunk 2]
```diff
@@ -106,6 +110,31 @@ async function updateAccountInfo (store: Store) {
...
+  if (accounts.length === 0) {
+    return
+  }
+  const apiProxy = getAPIProxy()
+  const { braveWalletService, keyringService, jsonRpcService } = apiProxy
```
Comment (josheleonard) [inline components/brave_wallet_ui/common/async/handlers.ts:120]: can be combined with line above

File: components/brave_wallet_ui/common/async/handlers.ts [hunk 3]
```diff
@@ -151,12 +180,23 @@ handler.on(WalletActions.accountsChanged.getType(), async (store) => {
   await updateAccountInfo(store)
 })
 
-// Will use this observer selectedAccountChanged action again once
-// selectedCoin is implemented here https://github.com/brave/brave-browser/issues/21465
+handler.on(WalletActions.selectNetwork.getType(), async (store: Store, payload: BraveWallet.NetworkInfo) => {
+  const jsonRpcService = getAPIProxy().jsonRpcService
```
Comment (josheleonard) [inline components/brave_wallet_ui/common/async/handlers.ts:184]: de-structuring preferred

File: components/brave_wallet_ui/common/wallet_api_proxy.ts [hunk 4]
```diff
@@ -62,8 +62,8 @@ export default class WalletApiProxy {
       autoLockMinutesChanged: function () {
         store.dispatch(WalletActions.autoLockMinutesChanged())
       },
-      selectedAccountChanged: function () {
-        store.dispatch(WalletActions.selectedAccountChanged())
+      selectedAccountChanged: function (coin) {
```
Comment (josheleonard) [inline components/brave_wallet_ui/common/wallet_api_proxy.ts:65]: add arg type?

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/common/async/handlers.ts (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/async/handlers.ts]: Combined!

File: components/brave_wallet_ui/common/async/handlers.ts (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/async/handlers.ts]: Combined!

File: components/brave_wallet_ui/common/async/handlers.ts (hunk 3 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/async/handlers.ts]: Deconstructed!

File: components/brave_wallet_ui/common/wallet_api_proxy.ts (hunk 4 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/wallet_api_proxy.ts]: Added type!

**Review by josheleonard — APPROVED** [review]

**Comment by srirambv** [comment]
Verification passed on <html><head></head>
<body>


Brave | 1.40.64 Chromium: 102.0.5005.50 (Official Build) nightly (64-bit)
-- | --
Revision | `f6e2cf8f59ec714bdcff8499991d55898875f287-refs/branch-heads/5005@{#648}`
OS | Windows 11 Version 22H2 (Build 22621.1)


</body>
</html>

- Verified steps from PR 
- Verified selecting account from panel automatically updates the account on the widget

https://user-images.githubusercontent.com/17010094/168959313-f74d9c90-d904-41ee-9b3c-18f8fdef9931.mov

**Pull request merged**